### PR TITLE
CodeQL PR

### PR DESCRIPTION
**Pull request by algojack**
## Summary

Looking at enabling CodeQL

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        languages: ${{ matrix.language }}
+        # If you wish to specify custom queries, you can do so here or in a config file.
+        # By default, queries listed here will override any specified in a config file.
+        # Prefix the list here with "+" to use these queries and those in the config file.
+        # queries: ./path/to/local/query, your-org/your-repo/queries@main
+
+    # Autobuild attempts to build any compiled languages  (C/C++, C#, or Java).
+    # If this step fails, then you should remove it and run the build manually (see below)
+    - name: Autobuild
+      uses: github/codeql-action/autobuild@v1
```
Comment (onetechnical) [inline .github/workflows/codeql-analysis.yml:57]: We may need to change this, or add dependencies to install boost-math.

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [rel/beta]
```
Comment (onetechnical) [inline .github/workflows/codeql-analysis.yml:16]: Saw warnings that we should add master here.

**Review by onetechnical — CHANGES_REQUESTED** [review]
This is largely just parroting the feedback from CodeQL itself. Does this mean we're actually _clean_ for CodeQL review right now?

File: .github/workflows/codeql-analysis.yml (hunk 2 above)
Comment (algojack) [reply .github/workflows/codeql-analysis.yml]: ah, i deleted one too many since i thought we are limited on minutes. I put it back. Thank you.

**Comment by algojack** [comment]
seems we need to do more things to make it work. Closing for now.

**Pull request closed without merging**